pkg/handlers/config.go

```diff
@@ -110,21 +169,18 @@ func UpdateAppConfig(w http.ResponseWriter, r *http.Request) {
 			}
 		}
 	}
+	updateAppConfigResponse.RequiredItems = requiredItems
```

Comment:
I think this should only be set if `len(requiredItems) > 0 && isPrimaryVersion`.

Reply:
no reason not to set it though
the calling code is currently dropping it, but that could change

Reply:
Oh, if the calling code is currently dropping it, that would cause a bug in the UI where a message indicating `This item is required` will not show up anymore next to required items upon saving. For that reason i suggested to move it inside that if block. we should also return this even if `createNewVersion` is true.

Reply:
the calling code is currently dropping the entire response struct
and returning the response struct generated by the isPrimaryVersion iteration only

Reply:
if this code is run, the caller does this:
```
	JSON(w, 200, UpdateAppConfigResponse{Success: true})
```